src/System.Private.CoreLib/src/System/Globalization/NumberFormatInfo.cs

```diff
@@ -346,14 +346,14 @@ public int[] CurrencyGroupSizes
             {
                 if (value == null)
                 {
-                    throw new ArgumentNullException("CurrencyGroupSizes",
+                    throw new ArgumentNullException(nameof(CurrencyGroupSizes),
                         SR.ArgumentNull_Obj);
                 }
                 Contract.EndContractBlock();
                 VerifyWritable();
 
                 Int32[] inputSizes = (Int32[])value.Clone();
-                CheckGroupSize("CurrencyGroupSizes", inputSizes);
+                CheckGroupSize(nameof(CurrencyGroupSizes), inputSizes);
```

Comment:
This looks fine. Yes, it would be nice to do a matching change in CoreCLR as well.